### PR TITLE
fix tests for SYCL target

### PR DESCRIPTION
**Pull request by jcosborn**

### CODE REVIEW

File: include/eigen_helper.h [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #define EIGEN_DONT_VECTORIZE
 #endif
 
-#include <cmath>
+#include <math.h>
```
Comment (maddyscientist) [inline include/eigen_helper.h:12]: why this change?

File: tests/invert_test_gtest.hpp [hunk 2]
```diff
@@ -205,6 +206,7 @@ INSTANTIATE_TEST_SUITE_P(MultiShiftEvenOdd, InvertTest,
                                  no_schwarz),
                          gettestname);
 
+#ifdef QUDA_MMA_AVAILABLE
```
Comment (maddyscientist) [inline tests/invert_test_gtest.hpp]: How about we skip this these tests rather than comment them out if MMA isn't available? e.g., `GTEST_SKIP`

File: lib/gauge_fix_ovr.cu [hunk 3]
```diff
@@ -71,7 +71,10 @@ namespace quda {
     bool advanceAux(TuneParam &param) const
     {
       param.aux.x = (param.aux.x + 1) % 6;
-       // mu must be contained in the block, types 0, 1, 2 have mu = 8 and 3, 4, 5 have mu = 4
+      if (!device::shared_memory_atomic_supported()) { // 1, 4 use shared memory atomics
```
Comment (maddyscientist) [inline lib/gauge_fix_ovr.cu:74]: Is lack of shared memory atomics a thing on some SYCL platforms?

**Review by maddyscientist — COMMENTED** [review]
Looks good.  Just a couple of minor queries, but we should be able to merge this pronto.

File: include/eigen_helper.h (hunk 1 above)
Comment (jcosborn) [reply include/eigen_helper.h:12]: Actually, it needs the functions (isinf, isnan) in the root namespace, not std::.

Comment (maddyscientist) [reply include/eigen_helper.h:12]: What needs the functions in the root namespace?

File: lib/gauge_fix_ovr.cu (hunk 3 above)
Comment (jcosborn) [reply lib/gauge_fix_ovr.cu:74]: This is mainly for testing.  I don't think there is actually a way to query it, so is seems to expect it to be supported.

File: include/eigen_helper.h (hunk 1 above)
Comment (jcosborn) [reply include/eigen_helper.h:12]: I think this is the cause: https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Core/MathFunctions.h#L984

File: tests/invert_test_gtest.hpp (hunk 2 above)
Comment (jcosborn) [reply tests/invert_test_gtest.hpp]: a3e633450853ed8c31bcc392632b69adb61cdf1e

**Review by maddyscientist — APPROVED** [review]
